SIAB/abacus/run.py

```
@@ -0,0 +1,344 @@
+
+# in the following, defines different tasks
+# -------------------------------------------#
+# TASK KIND1 - blscan                        #
+# DESCRIPTION: search bond lengths           #
+# -------------------------------------------#
+def blscan(general: dict,                  # general settings
+           calculation_setting: dict,      # calculation setting, for setting up INPUT file
+           env_settings: dict,             # calculation environment settings
+           reference_shape: str,           # reference shape, always to be dimer
+           nstep_bidirection: int = 5,     # number of steps for searching bond lengths per direction
+           stepsize: list = [0.2, 0.5],    # stepsize for searching bond lengths, unit in angstrom
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)